usaspending_api/database_scripts/job_archive/management/commands/generate_cares_act_test_data.py

```diff
@@ -0,0 +1,297 @@
+                    where   _base_submission_id is null;
+                """
+            )
+        )
+
+    def clone_submissions(self):
+        sql = self.read_sql_file("clone_submissions.sql")
+        for fiscal_period in self.clone_periods:
+            sqls = self.split_sql(
+                sql.format(
+                    submission_id_shift=fiscal_period * 100000000,  # to prevent id collisions,
+                    reporting_period_start=self.period_starts[fiscal_period],
+                    reporting_period_end=self.period_ends[fiscal_period],
+                    reporting_fiscal_period=fiscal_period,
+                    filter_fiscal_year=self.fiscal_year,
+                    filter_fiscal_period=self.fiscal_period,
+                    adjustment_ratio=self.period_ratios[fiscal_period],
+                )
+            )
+            self.run_sqls(sqls)
+
+    def update_submissions(self):
+        sql = self.read_sql_file("update_submissions.sql")
+        sqls = self.split_sql(
+            sql.format(
+                reporting_period_start=self.period_starts[self.fiscal_period],
+                reporting_period_end=self.period_ends[self.fiscal_period],
+                filter_fiscal_year=self.fiscal_year,
+                filter_fiscal_period=self.fiscal_period,
+                adjustment_ratio=self.period_ratios[self.fiscal_period],
+            )
+        )
+        self.run_sqls(sqls)
+
+    def clone_for_defc(self):
+        sql = self.read_sql_file("clone_for_defc.sql")
+        # These are just carefully selected "random" values used to clone File B and C records.
+        for p in [("I", 0.2, 13), ("F", 0.1, 11), ("L", 0.3, 10), ("M", 0.25, 8), ("N", 0.4, 7), ("O", 0.15, 5)]:
+            sqls = self.split_sql(
+                sql.format(
+                    disaster_emergency_fund_code=p[0],
+                    filter_fiscal_year=self.fiscal_year,
+                    filter_fiscal_quarter=self.fiscal_quarter,
+                    adjustment_ratio=p[1],
+                    divisor=p[2],
+                )
+            )
+            self.run_sqls(sqls)
+
+    def there_are_no_submissions(self):
+        return (
+            SubmissionAttributes.objects.filter(
+                reporting_fiscal_year=self.fiscal_year, reporting_fiscal_quarter=self.fiscal_quarter,
+            ).count()
+            == 0
+        )
+
+    def quarter_has_monthly_data(self):
+        return (
+            SubmissionAttributes.objects.filter(
+                reporting_fiscal_year=self.fiscal_year,
+                reporting_fiscal_quarter=self.fiscal_quarter,
+                quarter_format_flag=False,
+            ).count()
+            > 0
+        )
+
+    @staticmethod
+    def is_rds():
+        """ Not foolproof, but will hopefully prevent a few accidents between now and the end of CARES Act. """
+        return "rdsdbdata" in execute_sql_return_single_value("show data_directory")
```

Comment:
🤞  AWS doesn't move the data directory next week

Reply:
Yeah.  I want to find a better way to detect RDS instances.  My URI hack got squashed when we moved to SSH tunneling and I'm so afraid I'm going to run something important against an RDS instance by accident.  Like you said... hopefully this will carry us until the end of the CARES Act work.